snuba/migrations/migration.py

```diff
@@ -45,13 +46,64 @@ def backwards(self, context: Context, dry_run: bool) -> None:
         raise NotImplementedError
 
 
-class MultiStepMigration(Migration, ABC):
+class GlobalMigration(Migration, ABC):
+    """
+    Consists of one or more Python functions executed once globally.
+    """
+
+    @abstractmethod
+    def forwards_global(self) -> Sequence[RunPython]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def backwards_global(self) -> Sequence[RunPython]:
+        raise NotImplementedError
+
+    def forwards(self, context: Context, dry_run: bool) -> None:
+        if dry_run:
+            for op in self.forwards_global():
+                desc = op.description() or "No description provided"
+                print(f"Non SQL operation - {desc}")
+            return
+
+        migration_id, logger, update_status = context
+        logger.info(f"Running migration: {migration_id}")
+        if not self.is_first_migration():
```

Comment:
Is the initial migration supposed to be a node migration or a global migration?

Reply:
It's a node migration, just copied it over for "completeness" I guess (though it's probably not quite right). On second thoughts, I'm going to drop this from here since it's never used. I actually want to change the first migration implementation to use a special InitialMigration subclass of node migration instead, but will tackle that separately.